File: DataStructures/LinkedList/pointersNthFromEnd.py
class Node: 

    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:

    def __init__(self):
        self.head = None

    def push(self, new_data):
         new_node = Node(new_data)
         new_node.next = self.head
         self.head = new_node

    def printNthFromLast(self, n):
        main_ptr = self.head
        ref_ptr = self.head

        count = 0
        if(self.head is not None):
            while(count < n):
                ref_ptr = ref_ptr.next
                count+=1
            if(ref_ptr is None):
                print("%d is greater than the no of nodes",n)
                return
        if(ref_ptr is None):
            self.head = self.head.next
            if(self.head is not None):
                print "Node no %d from the last is %d" %(n, main_ptr.data)

            while((ref_ptr is not None) and (ref_ptr.next is not None)):
                main_ptr = main_ptr.next
                ref_ptr = ref_ptr.next

            print("Node no % d from the last is % d" %(n, main_ptr.data))
if __name__ == "__main__":
    llist = LinkedList()
    llist.push(20)
    llist.push(25)
    llist.push(4)
    llist.push(35)
    llist.push(1)

    llist.printNthFromLast(4)

File: DataStructures/LinkedList/pointersNthFromEnd.java
class LinkedList 
{
    Node head;
    LinkedList(int d){Node head = new Node(d)};
    // I would do this 

    class Node {
        int data;
        Node next;
        Node (int d)
        {
            data = d;
            next = null;
        }
    }

    void printNthFromLast(int n)
    {
        Node main_ptr = head;
        Node ref_ptr = head;

        int count = 0;
        if(head != null)
        {
            while (count < n)
            {
                if(ref_ptr == null)
                {
                    System.out.print(n + " is greater than the no " + " of nodes in the list");
                    return;
                }
                ref_ptr = ref_ptr.next;
                count++;
            }
            if(ref_ptr == null)
            {
                if(head != null)
                {
                    System.out.print("Node no. " + n + " from the last is " + head.data); //this is technically not true but is in the code
                }
                else{
                    while(ref_ptr != null)
                    {
                        main_ptr = main_ptr.next;
                        ref_ptr = ref_ptr.next;
                    }
                    System.out.print("Node no. " + n + " from the last is " + main_ptr.data);
                }
            }
        }
    }


    public void push(int new_data)
    {
        Node new_node = new Node(new_data);
        new_node.next = head;
        head = new_node.next;
    }

    public static void main (String[] args)
    {
        LinkedList llist = new LinkedList();
        llist.push(20);
        llist.push(4);
        llist.push(15);
        llist.push(35);

        llist.printNthFromLast(4);
    }
}